crates/common/sync/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+use alloy_primitives::B256;
+use anyhow::{anyhow, ensure};
+use checkpoint::fetch_default_checkpoint_url;
+use ream_consensus::{
+    blob_sidecar::{BlobIdentifier, BlobSidecar},
+    constants::SECONDS_PER_SLOT,
+    electra::{beacon_block::SignedBeaconBlock, beacon_state::BeaconState},
+    execution_engine::rpc_types::get_blobs::BlobAndProofV1,
+    misc::{compute_epoch_at_slot, compute_start_slot_at_epoch},
+};
+use ream_fork_choice::{
+    handlers::on_tick,
+    store::{Store, get_forkchoice_store},
+};
+use ream_network_spec::networks::network_spec;
+use ream_rpc::types::response::BeaconVersionedResponse;
+use ream_storage::{db::ReamDB, tables::Table};
+use tracing::{info, warn};
+pub mod checkpoint;
+
+/// Entry point for checkpoint sync.
+pub async fn initialize_db_from_checkpoint(
+    db: ReamDB,
+    rpc: Option<String>,
+) -> anyhow::Result<(Store, u64)> {
+    let checkpoint_sync_url = match rpc {
+        Some(url) => url,
+        // 0 for Mainnet, 1 for Sepolia
+        None => fetch_default_checkpoint_url().expect("Unable to fetch default checkpoint url")[1]
+            .checkpoint_urls[0]
+            .clone(),
+    };
```

Comment:
```suggestion
    let checkpoint_sync_url = match rpc {
        Some(checkpoint_sync_url) => checkpoint_sync_url,
        None => fetch_default_checkpoint_url().expect("Unable to fetch default checkpoint url")[1]
            .checkpoint_urls[0]
            .clone(),
    };
```

We should be using the network spec to get which network it is and


You can do something like we did for

```
impl Bootnodes {
    pub fn to_enrs(self, network: Network) -> Vec<Enr> {
```